authentication.py

```diff
@@ -0,0 +1,92 @@
+import re
+from passlib.context import CryptContext
+from fastapi import HTTPException, status
+import jwt
+
+from models import User
+from config import get_settings
+
+pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
+
+
+def get_hashed_password(password):
+    return pwd_context.hash(password)
+
+
+async def very_token(token: str):
+    '''verify token from login'''
+    try:
+        payload = jwt.decode(token, get_settings().SECRET,
+                             algorithms=["HS256"])
+        user = await User.get(id=payload.get("id"))
+
+    except:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Invalid Token",
+            headers={"WWW-Authenticate": "Bearer"}
+        )
+    return await user
+
+
+async def very_token_email(token: str):
+    '''verify token from email'''
+    try:
+        payload = jwt.decode(token, get_settings().SECRET,
+                             algorithms=["HS256"])
+        user = await User.get(id=payload.get("id"), email=payload.get("email"))
+
+    except:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Invalid Token",
+            headers={"WWW-Authenticate": "Bearer"}
+        )
+    return await user
+
+# bug: sam_ple@gma.com:Trye ; sam_p_le@gma.com: False!!
+regex = '^[a-z0-9]+[\._]?[a-z0-9]+[@]\w+[.]\w{2,3}$'
+
+
+def is_not_email(email):
+    """if valid mail: return 'True' \n 
+     ** This is a simple way to do this and is not recommended for a real project ** """
+    if(re.search(regex, email)):
```

Comment:
**Performance Issue**: The regex used for email validation is compiled every time the is_not_email function is called. This is inefficient, especially if the function is called frequently. <br> **Fix**: Compile the regex pattern once and reuse it, instead of compiling it every time the function is called. <br> **Code Suggestion**: 
 ```
 @@ -47,3 +47,4 @@
 -regex = '^[a-z0-9]+[\\._]?[a-z0-9]+[@]\\w+[.]\\w{2,3}$'
 +email_pattern = re.compile('^[a-z0-9]+[\\._]?[a-z0-9]+[@]\\w+[.]\\w{2,3}$')
 @@ -54,2 +54,2 @@
 -    if(re.search(regex, email)):
 +    if(email_pattern.search(email)):
 ```